Source/AliveLibAO/Abe.cpp

```diff
@@ -5699,7 +5699,146 @@ void Abe::State_34_RunJumpLand_427560()
 
 void Abe::State_35_RunLoop_425060()
 {
-    NOT_IMPLEMENTED();
+    field_10C_prev_held |= sInputObject_5009E8.field_0_pads[sCurrentControllerIndex_5076B8].field_0_pressed;
+
+    Event_Broadcast_417220(kEventNoise_0, this);
+    Event_Broadcast_417220(kEvent_10, this);
+
+    if (WallHit_401930(field_BC_sprite_scale * FP_FromInteger(50), field_B4_velx))
+    {
+        field_2A8_flags.Clear(Flags_2A8::e2A8_Bit3);
+        ToKnockback_422D90(1, 1);
+        return;
+    }
+
+    sub_422FC0();
+
+    if (field_FC_current_motion != eAbeStates::State_35_RunLoop_425060)
+    {
+        field_2A8_flags.Clear(Flags_2A8::e2A8_Bit3);
+        return;
+    }
+
+    if (field_10_anim.field_92_current_frame && field_10_anim.field_92_current_frame != 8)
```

Comment:
this looks suspect, should be != 0 I think?

Reply:
nope it's actually supposed to be like that

Reply:
!= 0 is the same effect but more readable though